pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -154,6 +150,10 @@ func (p *Planner) ConfigureFetch() plan.FetchConfiguration {
 		},
 		Variables:            p.variables,
 		DisallowSingleFlight: p.disallowSingleFlight,
+		ProcessResponseConfig: resolve.ProcessResponseConfig{
+			ExtractGraphqlResponse:    true,
```

Comment:
what happens if this is false? shouldn't it be the default?

Reply:
it is always false for all other datasources

I was choosing between 2 approaches `NonGraphqlResponse` and `GraphqlResponse`

so it is like set to `false` for all other and future data sources or set `true` for the graphql data source

Reply:
what's the expectation on "non graphql" datasources? What format should they return?
wouldn't it be easier if they just return {"errors":null,"data":null} format as well and we treat it like a standard?